src/modules/pages/HealthCheckPage/HealthCheckList.js

```diff
@@ -0,0 +1,105 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import R from 'ramda';
+import {
+  Link,
+} from 'react-router-dom';
+
+import block from '../../../helpers/bem-cn';
+
+import Correct from './Correct';
+import Icon from '../../Icon/Icon';
+import Pagination from '../../Pagination/Pagination';
+import HealthCheckModal from '../../modals/HealthCheckModal/HealthCheckModal';
+
+import './HealthCheckList.css';
+
+const b = block('j-healthcheck');
+const bList = block(b('list')());
+const bItem = block(b('list-item')());
+
+const propTypes = {
+    clearHealthCheckDetails: PropTypes.func.isRequired,
+    currentPageIndex: PropTypes.number.isRequired,
+    fetchHealthCheckList: PropTypes.func.isRequired,
+    fetchHealthCheckItem: PropTypes.func.isRequired,
+    healthcheckList: PropTypes.arrayOf(PropTypes.object).isRequired,
+    setCurrentPageIndex: PropTypes.func.isRequired,
+};
+
+class HealthCheckList extends PureComponent {
+    componentDidMount() {
+        this.props.fetchHealthCheckList();
+    }
+
+    test = name => {
+        this.props.fetchHealthCheckItem(name);
+    }
+
+    renderList = () => {
+        return (
+            <div>Oooops</div>
+        );
+    }
+
+    renderHealthcheckInfo = list => {
+        const {
+            clearHealthCheckDetails,
+            problemToDisplay,
+            status,
+        } = this.props;
+
+        if (status) {
+            return (
+                <div className={b('pane').mix('j-pane')}>
+                    <Correct className={b('correct')()} />
+                </div>
+            );
+        }
+
+        return (
+            <div className={bList()}>
+                {
+                    list.map(item => {
+                        return (
+                            <div className={bItem()} key={item.name}>
+                                <div className={bItem('name')}>{item.name}</div>
+                                <div className={bItem('message')}>{item.message}</div>
+                                <div className={bItem('details')} onClick={() => this.test(item.name)}>Show Details</div>
+                                <Link to={`/${item.name}`}>
+                                    <Icon type="edit" />
+                                </Link>
+                            </div>
+                        );
+                    })
+                }
+
+                <HealthCheckModal
+                    className={b('modal')()}
+                    isOpen={!R.isEmpty(problemToDisplay)}
```

Comment:
on this page **yes**, only for this. Imho this is way faster and shorter than manually check for `null`, `undefined` and existence of properties.